### PR TITLE
Added new policies + updated GKE CIS version to 1.4

### PR DESCRIPTION
**Pull request by mikouaj**
* Added new policy: `Control plane user basic authentication`
* Added new policy: `Control plane user certificate authentication`
* Added new policy: `Customer-Managed Encryption Keys for persistent disks`
* Added new policy: `Enable Security Posture dashboard`
* Added new policy: `Enable Workload vulnerability scanning`
* Added CIS benchmark for policy: `GKE VPC-native cluster`
* Corrected CIS benchmark for policy: `Use Container - Optimised OS`
* Updated version of GKE CIS to 1.4
* Ported `GKE intranode visibility` policy to policies v1 directory

### CODE REVIEW

File: gke-policies-v2/policy/cluster_enable_security_posture.rego [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+default valid := false
+
+valid {
+  count(violation) == 0
+}
+
+violation[msg] {
+  not input.data.gke.security_posture_config.mode == 2
+  msg := "GKE cluster has not disabled Security Posture"
```
Comment (ewojtach) [inline gke-policies-v2/policy/cluster_enable_security_posture.rego:41]: has not enabled?

File: gke-policies-v2/policy/cluster_enable_workload_scanning_test.rego [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }}}
+}
+
+test_cluster_disabled_workload_scanning {
+    not valid with input as {"data": {"gke": {
+        "name": "cluster-test", 
+        "security_posture_config": {
+           "mode": 1,
+           "vulnerability_mode": 0
```
Comment (ewojtach) [inline gke-policies-v2/policy/cluster_enable_workload_scanning_test.rego:32]: shouldn't be 1 for disabled? 
this and the next one tests are exactly the same

File: gke-policies-v2/policy/control_plane_disable_cert_authentication.rego [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# title: Control plane user certificate authentication
+# description: >-
+#   Disable Client Certificates, which require certificate rotation, for authentication. Instead,
+#   use another authentication method like OpenID Connect.
+# custom:
+#   group: Security
+#   severity: High
+#   recommendation: >
+#     Client certificate authentication cannot be disabled on the existing cluster.
+#     The new cluster has to be created with a "Clinet certificate" option disabled.
```
Comment (ewojtach) [inline gke-policies-v2/policy/control_plane_disable_cert_authentication.rego:25]: Client - typo

File: gke-policies-v2/policy/control_plane_disable_cert_authentication.rego [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+default valid := false
+
+valid {
+	count(violation) == 0
+}
+
+violation[msg] {
+	input.data.gke.master_auth.client_certificate
+	msg := "The GKE cluster authentication is configured with a client certificate"
```
Comment (ewojtach) [inline gke-policies-v2/policy/control_plane_disable_cert_authentication.rego:43]: maybe we should emphasize in the message what is wrong with it? eg:
The GKE cluster authentication should not be configured with a client certificate

File: gke-policies-v2/policy/control_plane_disable_cert_authentication.rego [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+violation[msg] {
+	input.data.gke.master_auth.client_key
+	msg := "The GKE cluster authentication is configured with a client key"
```
Comment (ewojtach) [inline gke-policies-v2/policy/control_plane_disable_cert_authentication.rego:48]: The GKE cluster authentication should not be configured with a client key?

File: gke-policies-v2/policy/control_plane_disable_password_authentication.rego [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+default valid := false
+
+valid {
+	count(violation) == 0
+}
+
+violation[msg] {
+	input.data.gke.master_auth.password
+	msg := "The GKE cluster authentication is configured with a client password"
```
Comment (ewojtach) [inline gke-policies-v2/policy/control_plane_disable_password_authentication.rego:44]: The GKE cluster authentication should not be configured with a client password ?

File: gke-policies-v2/policy/control_plane_disable_password_authentication.rego [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+violation[msg] {
+	input.data.gke.master_auth.username
+	msg := "The GKE cluster authentication is configured with a client username"
```
Comment (ewojtach) [inline gke-policies-v2/policy/control_plane_disable_password_authentication.rego:49]: The GKE cluster authentication should not be configured with a client username

File: gke-policies/policy/cluster_enable_security_posture.rego [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+default valid := false
+
+valid {
+  count(violation) == 0
+}
+
+violation[msg] {
+  not input.security_posture_config.mode == 2
+  msg := "GKE cluster has not disabled Security Posture"
```
Comment (ewojtach) [inline gke-policies/policy/cluster_enable_security_posture.rego:41]: enabled?

File: gke-policies/policy/cluster_enable_workload_scanning_test.rego [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }
+}
+
+test_cluster_disabled_workload_scanning {
+    not valid with input as {
+        "name": "cluster-test", 
+        "security_posture_config": {
+           "mode": 1,
+           "vulnerability_mode": 0
```
Comment (ewojtach) [inline gke-policies/policy/cluster_enable_workload_scanning_test.rego:32]: vulnerability_mode:1?

File: gke-policies/policy/control_plane_disable_cert_authentication.rego [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+# title: Control plane user certificate authentication
+# description: >-
+#   Disable Client Certificates, which require certificate rotation, for authentication. Instead,
+#   use another authentication method like OpenID Connect.
+# custom:
+#   group: Security
+#   severity: High
+#   recommendation: >
+#     Client certificate authentication cannot be disabled on the existing cluster.
+#     The new cluster has to be created with a "Clinet certificate" option disabled.
```
Comment (ewojtach) [inline gke-policies/policy/control_plane_disable_cert_authentication.rego:25]: Client

File: gke-policies/policy/control_plane_disable_cert_authentication.rego [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+default valid := false
+
+valid {
+	count(violation) == 0
+}
+
+violation[msg] {
+	input.master_auth.client_certificate
+	msg := "The GKE cluster authentication is configured with a client certificate"
```
Comment (ewojtach) [inline gke-policies/policy/control_plane_disable_cert_authentication.rego:43]: maybe:
The GKE cluster authentication should not be configured with...?

File: gke-policies/policy/control_plane_disable_cert_authentication.rego [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+violation[msg] {
+	input.master_auth.client_key
+	msg := "The GKE cluster authentication is configured with a client key"
```
Comment (ewojtach) [inline gke-policies/policy/control_plane_disable_cert_authentication.rego:48]: maybe:
The GKE cluster authentication should not be configured with...?

File: gke-policies/policy/control_plane_disable_password_authentication.rego [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+default valid := false
+
+valid {
+	count(violation) == 0
+}
+
+violation[msg] {
+	input.master_auth.password
+	msg := "The GKE cluster authentication is configured with a client password"
```
Comment (ewojtach) [inline gke-policies/policy/control_plane_disable_password_authentication.rego:44]: maybe:
The GKE cluster authentication should not be configured with...?

File: gke-policies/policy/intranode_visibility.rego [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+default valid := false
+
+valid {
+	count(violation) == 0
+}
+
+violation[msg] {
+	not input.networkConfig.enableIntraNodeVisibility = true
+
+	msg := "The GKE cluster does not Intranode Visibility enabled"
```
Comment (ewojtach) [inline gke-policies/policy/intranode_visibility.rego:43]: shouldn't it be like:
The GKE cluster does not have Intranode Visibility enabled ?

and probably the same for policies v2?

**Review by ewojtach — APPROVED** [review]
LGTM! just few typos and rephrazing ideas

File: gke-policies-v2/policy/cluster_enable_security_posture.rego (hunk 1 above)
Comment (mikouaj) [reply gke-policies-v2/policy/cluster_enable_security_posture.rego]: 👍 Fixed

File: gke-policies-v2/policy/cluster_enable_workload_scanning_test.rego (hunk 2 above)
Comment (mikouaj) [reply gke-policies-v2/policy/cluster_enable_workload_scanning_test.rego]: You are right, thank you!

File: gke-policies-v2/policy/control_plane_disable_cert_authentication.rego (hunk 5 above)
Comment (mikouaj) [reply gke-policies-v2/policy/control_plane_disable_cert_authentication.rego]: Thats better, thanks !

File: gke-policies/policy/control_plane_disable_cert_authentication.rego (hunk 10 above)
Comment (mikouaj) [reply gke-policies/policy/control_plane_disable_cert_authentication.rego]: 👍 , fixed!

File: gke-policies/policy/control_plane_disable_cert_authentication.rego (hunk 12 above)
Comment (mikouaj) [reply gke-policies/policy/control_plane_disable_cert_authentication.rego]: Sounds better, thanks!

File: gke-policies/policy/intranode_visibility.rego (hunk 14 above)
Comment (mikouaj) [reply gke-policies/policy/intranode_visibility.rego]: Yes I agree - I rephrased it in both policies v1 and v2, thanks!

File: gke-policies-v2/policy/control_plane_disable_cert_authentication.rego (hunk 4 above)
Comment (mikouaj) [reply gke-policies-v2/policy/control_plane_disable_cert_authentication.rego]: Sounds better, thanks!

File: gke-policies/policy/control_plane_disable_cert_authentication.rego (hunk 11 above)
Comment (mikouaj) [reply gke-policies/policy/control_plane_disable_cert_authentication.rego]: +1, fixed!

File: gke-policies/policy/control_plane_disable_password_authentication.rego (hunk 13 above)
Comment (mikouaj) [reply gke-policies/policy/control_plane_disable_password_authentication.rego:44]: Sounds better, fixed !

File: gke-policies/policy/cluster_enable_workload_scanning_test.rego (hunk 9 above)
Comment (mikouaj) [reply gke-policies/policy/cluster_enable_workload_scanning_test.rego]: +1, fixed!

File: gke-policies/policy/cluster_enable_security_posture.rego (hunk 8 above)
Comment (mikouaj) [reply gke-policies/policy/cluster_enable_security_posture.rego]: +1, fixed!

File: gke-policies-v2/policy/control_plane_disable_password_authentication.rego (hunk 7 above)
Comment (mikouaj) [reply gke-policies-v2/policy/control_plane_disable_password_authentication.rego]: +1, fixed!

File: gke-policies-v2/policy/control_plane_disable_password_authentication.rego (hunk 6 above)
Comment (mikouaj) [reply gke-policies-v2/policy/control_plane_disable_password_authentication.rego]: +1, fixed

**Pull request merged**